src/test/java/org/quiltmc/launchermeta/version_manifest/VersionManifestTest.java

```diff
@@ -42,17 +44,44 @@ public class VersionManifestTest {
                 }]
             }
             """;
+    private static final String TEST_JSON_V2 = """
+            {
+                "latest": {
+                    "release": "1.17.1",
+                    "snapshot": "21w42a"
+                },
+                "versions": [{
+                    "id": "21w42a",
+                    "type": "snapshot",
+                    "url": "https://piston-meta.mojang.com/v1/packages/3ce8fdf60e69bfb0944e479ada4cf6b60dcc3995/21w42a.json",
+                    "time": "2021-10-20T12:46:24+00:00",
+                    "releaseTime": "2021-10-20T12:41:25+00:00",
+                    "sha1": "3ce8fdf60e69bfb0944e479ada4cf6b60dcc3995",
+                    "complianceLevel": 1
+                }]
+            }
+            """;
 
     @Test
     public void testParseFullJson() throws IOException {
-        JsonElement json = TestUtil.getJsonFromURL(VersionManifestTest.MANIFEST_URL);
-        VersionManifest manifest = VersionManifest.fromJson(json);
+        checkRemoteMatches(MANIFEST_URL, null);
+    }
 
-        assertEquals(manifest.getVersions().size(), json.getAsJsonObject().get("versions").getAsJsonArray().size(), "Size of version array matches json");
-        assertEquals(manifest.getLatestVersions().getRelease(), json.getAsJsonObject().get("latest").getAsJsonObject().get("release").getAsString(), "Size of version array matches json");
+    @Test
+    public void testParseFullJsonV2() throws IOException {
+        checkRemoteMatches(MANIFEST_URL_V2, (json, manifest) -> {
+            String latestVersion = manifest.getLatestVersions().getRelease();
+            VersionEntry latestEntry = manifest.getVersions().stream()
+                    .filter(entry -> entry.getId().equals(latestVersion))
+                    .findFirst()
+                    .orElseThrow();
+            assertNotNull(latestEntry.getSha1(), "Latest version has a sha1");
```

Comment:
Those two lines don't really add anything of value to the test, I was wondering about replacing them with something along the lines of:

```java
JsonArray versions = json.getAsJsonObject().get("versions").getAsJsonArray();
for (JsonElement version : versions) {
    if (version.getAsJsonObject().get("id").getAsString().equals(latestVersion)) {
        assertEquals(version.getAsJsonObject().get("sha1").getAsString(), latestEntry.getSha1().orElseThrow(), "Latest version the right sha1");
        assertEquals(version.getAsJsonObject().get("complianceLevel").getAsInt(), latestEntry.getComplianceLevel(), "Latest version the right compliance level");
        return;
    }
}
```

but I'm worrying that making the test too dependent on "unstable" factors such as the remote manifest JSON might be bad-practice of some sort.